### PR TITLE
Feat: add "retry/workflow" endpoint

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [VADC-519](https://ctds-planx.atlassian.net/browse/VADC-519)

### New Features
- New retry endpoint to retry a failed workflow and some minor improvements in nearby code

### Improvements
- Updated Api docs

[VADC-519]: https://ctds-planx.atlassian.net/browse/VADC-519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kmhernan — CHANGES_REQUESTED** [review]
Need to discuss

File: src/argowrapper/engine/argo_engine.py [hunk 1]
```diff
@@ -177,6 +187,49 @@ def cancel_workflow(self, workflow_name: str) -> string:
...
+            # Call the archived retry (will raise NotFoundException if workflow is not yet archived):
+            self.archive_api_instance.retry_archived_workflow(
+                uid=uid,
+                body=IoArgoprojWorkflowV1alpha1RetryArchivedWorkflowRequest(
+                    _check_type=False,
+                ),
+                _check_return_type=False,
+            )
+            return f"archived {workflow_name} retried sucessfully"
+        except NotFoundException:
```
Comment (kmhernan) [inline src/argowrapper/engine/argo_engine.py:213]: We will probably need to revisit this for other exceptions

**Review by kmhernan — APPROVED** [review]
Approved, although I think you can only "retry" failed workflows, I don't think we check that but it'd probably throw an exception

**Comment by pieterlukasse** [comment]
@kmhernan that's a good point. Argo returns this: http 400,  {"code":3,"message":"workflow must be Failed/Error to retry"}

**Pull request merged**